### ISSUE TITLE
ui too big to reset

### ISSUE THREAD

**Issue by Reaper10** [hydrogen-music/hydrogen#1933]
Hydrogen version *          :  1.2.3
Operating system + version  :  wIn 10 22H2
Audio driver + version      : 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
ui too big to reset.
<img width="831" alt="Untitled" src="https://github.com/hydrogen-music/hydrogen/assets/8137561/21f6f8f8-0fc4-400f-81ea-082f5c25ac2d">
as you cam see I can't get to the olay button

**Comment by theGreatWhiteShark** [comment]
I see. But this look more like a scaling issue of your OS and not like a Hydrogen bug.

In Windows > Settings > Display > Scale and layout you probably set a value larger than 100%, right? Hydrogen will comply with it and will be rendered larger.

When starting it from the Windows CMD.exe prompt you can also fine tune the scaling size (`1.0` mean no scaling)

``` cmd
set QT_SCALE_FACTOR=0.9
start /b "" "C:\Program Files\Hydrogen\hydrogen.exe"
```

**Comment by theGreatWhiteShark** [comment]
See #1347

**Comment by Reaper10** [comment]
> I see. But this look more like a scaling issue of your OS and not like a Hydrogen bug.
> 
> In Windows > Settings > Display > Scale and layout you probably set a value larger than 100%, right? Hydrogen will comply with it and will be rendered larger.
> 
> When starting it from the Windows CMD.exe prompt you can also fine tune the scaling size (`1.0` mean no scaling)
> 
> ```batchfile
> set QT_SCALE_FACTOR=0.9
> start /b "" "C:\Program Files\Hydrogen\hydrogen.exe"
> ```

scale is 150% am I supid for have set this way?

**Comment by cme** [comment]
I think it's definitely a bit of a usability bug to have elements go off-screen in a way that they can't be accessed. I'm sure there should be a standard way of handling oversized dialogs (adding scrollbars to the window when it's too big?) but I don't know off the top of my head. Hm.

**Comment by theGreatWhiteShark** [comment]
> I'm sure there should be a standard way of handling oversized dialogs (adding scrollbars to the window when it's too big?) but I don't know off the top of my head. Hm.

Something like restricting the window height and width to 100%? Hmm. Not sure. In browsers there is at least a somewhat standardized way for those measures but I would be surprised if all window managers all for that. And if you have multiple screens and move the application around or mirror it at different screen sizes/resolutions? Maybe some users even want to expand it to cover multiple screens in some uses-cases.

> scale is 150% am I supid for have set this way?

So, the issue is resolved at your end? (In case you find labels harder to read with a smaller scale factor, you can also increase the font size in the Preferences)

**Comment by Reaper10** [comment]
no its still too big
<img width="840" alt="Screenshot 2024-05-22 202855" src="https://github.com/hydrogen-music/hydrogen/assets/8137561/277c3701-f059-4a77-b96f-94d5fcd694c0">

**Comment by theGreatWhiteShark** [comment]
> no its still too big

even with your Windows > Settings > Display > Scale set to 100% again?

**Comment by Reaper10** [comment]
no 150%

**Comment by theGreatWhiteShark** [comment]
> no 150%

Ok. So, when set back to `100%` and after setting "High-resolution display scaling" in Preferences > Appearance > Interface to `Prefer smaller` Hydrogen still does not fit your display?

**Comment by Reaper10** [comment]
i need the rest of the ui at 150

**Comment by theGreatWhiteShark** [comment]
> i need the rest of the ui at 150

Ah, now I get it! You do prefer the GUI to be larger but Hydrogen is well suited for this size as it just overflows instead of providing e.g. scrollbars.

I thought you wanted it smaller and it was big on accident. Sorry it took so long.

**Comment by theGreatWhiteShark** [comment]
I would propose this solution: 

Hydrogen will be able to shrink below its current minimal size. Whenever this happens scrollbars for the whole application will appear. So, you could make Hydrogen fit your screen and still access parts that don't fit into it.

![2024-06-03-180937_3840x1200_scrot](https://github.com/hydrogen-music/hydrogen/assets/14164547/08f3df1d-01b6-47d0-8f60-05390008ff94)

That's not ideal but it's feasible.  @Reaper10 what do you think?

**Comment by Reaper10** [comment]
im not a programmer so i speak on but as a uer look good

**Comment by theGreatWhiteShark** [comment]
Hey @Reaper10 ,

Could you try [this version of Hydrogen](https://ci.appveyor.com/api/buildjobs/p7e9765g3h0fw4fw/artifacts/Hydrogen-1.2.3-172-g14ab85a3f-win64.exe) and check whether the size issue is still present?

Do you have problems with other parts of the application besides the main window as well?

**Comment by Reaper10** [comment]
still can't ok button 
<img width="840" alt="Screenshot 2024-06-18 194313" src="https://github.com/hydrogen-music/hydrogen/assets/8137561/e326f61c-512b-49ef-a1e1-34c83fe8ad42">

**Comment by theGreatWhiteShark** [comment]
> still can't ok button

Damn. I noted that too using large scaling factors but hoped your's would be smaller and I do not have to touch these parts of the code. 😬 

But it's probably best to introduce scrollbars for all big windows. Alright.

Apart from the Preferences dialog I also noted that 
- Mixer
- Sample Editor
- Playlist Editor

are too large as well. Anything else?

**Comment by Reaper10** [comment]
the rest of the ui with this

> > still can't ok button
> 
> Damn. I noted that too using large scaling factors but hoped your's would be smaller and I do not have to touch these parts of the code. 😬
> 
> But it's probably best to introduce scrollbars for all big windows. Alright.
> 
> Apart from the Preferences dialog I also noted that
> 
> * Mixer
> * Sample Editor
> * Playlist Editor
> 
> are too large as well. Anything else?

the rest of the ui with this works great

**Comment by theGreatWhiteShark** [comment]
I ended up tweaking some more dialogs, like drumkit properties and online import or the one for audio file loading, as well. Could you check whether [this version](https://ci.appveyor.com/api/buildjobs/8ihg05nwqpa9i098/artifacts/Hydrogen-1.2.3-187-g67e0dfd95-win64.exe) works fine? Or are there still any inaccessible parts?

**Comment by Reaper10** [comment]
> I ended up tweaking some more dialogs, like drumkit properties and online import or the one for audio file loading, as well. Could you check whether [this version](https://ci.appveyor.com/api/buildjobs/8ihg05nwqpa9i098/artifacts/Hydrogen-1.2.3-187-g67e0dfd95-win64.exe) works fine? Or are there still any inaccessible parts?

it great i have't tried the sample editor
